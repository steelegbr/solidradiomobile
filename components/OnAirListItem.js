/**
    Solid Radio Mobile App
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import React from "react";
import { Component } from "react";
import { connect } from "react-redux";
import { List } from "react-native-paper";
import { generateStationTheme } from "../branding/branding";

/**
 * List item for displaying the currently on air show.
 */

class OnAirListItem extends Component {

    render() {

        // Read in the properties

        const { onAir } = this.props;

        // Render the card

        return (
            <List.Item
                title={`Now On Air - ${onAir.show}`}
                description={onAir.description}
                left={props => <List.Icon icon={{ uri: onAir.image }} />}
            />
        );

    }

}

function mapStateToProps(state, ownProps) {

    return {
        onAir: ownProps.station.onAir
    };

}

export default connect(mapStateToProps)(OnAirListItem);