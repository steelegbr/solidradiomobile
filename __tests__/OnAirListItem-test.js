/**
 * Tests the on air list item.
 */

import React from 'react';
import OnAirListItem from '../components/OnAirListItem';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

// Redux

const mockStore = configureStore([]);
let store;

beforeEach(() => {
    store = mockStore({});
});

// Tests

it('renders', () => {

    const station ={ 
        onAir: {
            show: "Test Show",
            description: "A really good test description.",
            image: "art.png"
        }
    };

    const onAirItem = renderer.create(
        <Provider store={store}>
            <OnAirListItem station={station} />
        </Provider>
    ).toJSON();
    expect(onAirItem).toMatchSnapshot();

});
