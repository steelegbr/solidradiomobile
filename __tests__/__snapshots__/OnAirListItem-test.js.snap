// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "margin": 8,
            "width": 40,
          },
          undefined,
        ]
      }
    >
      <Image
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        source={
          Object {
            "uri": "art.png",
          }
        }
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "height": 24,
              "resizeMode": "contain",
              "tintColor": undefined,
              "width": 24,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginVertical": 6,
            "paddingLeft": 8,
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
              },
              undefined,
            ],
          ]
        }
      >
        Now On Air - Test Show
      </Text>
      <Text
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "fontSize": 14,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
              },
              undefined,
            ],
          ]
        }
      >
        A really good test description.
      </Text>
    </View>
  </View>
</View>
`;
